prov/shm/src/smr.h

```diff
@@ -280,6 +281,8 @@ struct smr_ep {
 	struct smr_queue	unexp_msg_queue;
 	struct smr_queue	unexp_tagged_queue;
 	struct dlist_entry	sar_list;
+	/* cache for use with hmem ipc */
+	struct ofi_mr_cache	*ipc_cache;
```

Comment:
This looks like the wrong place for a cache.  A cache per fabric or domain object make more sense, not per endpoint.  Even global might be needed for this.  I think other caches end up being per domain.